great_expectations/expectations/metrics/util.py

```diff
@@ -343,7 +343,9 @@ def get_sqlalchemy_column_metadata(
                     columns = table_selectable.columns().columns
             else:
                 columns = inspector.get_columns(
-                    table_selectable,
+                    str(
+                        table_selectable
+                    ),  # TODO: remove cast to a string once [this](https://github.com/snowflakedb/snowflake-sqlalchemy/issues/157) issue is resovled
```

Comment:
```suggestion
                    # TODO: remove cast to a string once [this](https://github.com/snowflakedb/snowflake-sqlalchemy/issues/157) issue is resolved
                    table_selectable if engine.dialect.name != "snowflake" else str(table_selectable),
```

We can pull out the dialect name from the `engine` and then only do this change for `snowflake`.
That way we limit the blast radius of this change.

Reply:
I want to avoid adding conditionals unless there's a clear need. Based on the tests we have, the conversion here does not seem to be negatively affecting the test cases